src/client/Client.cc

```diff
@@ -11334,43 +11334,7 @@ int Client::ll_lookup_inode(
   ldout(cct, 3) << "ll_lookup_inode " << ino  << dendl;
 
   // Num1: get inode and *inode
-  int r = _lookup_ino(ino, perms, inode);
-  if (r)
-    return r;
-
-  ceph_assert(*inode != NULL);
-
-  if (!(*inode)->dentries.empty()) {
-    ldout(cct, 8) << __func__ << " dentry already present" << dendl;
-    return 0;
-  }
-
-  if ((*inode)->is_root()) {
-    ldout(cct, 8) << "ino is root, no parent" << dendl;
-    return 0;
-  }
-
-  // Num2: Request the parent inode, so that we can look up the name
-  Inode *parent;
-  r = _lookup_parent(*inode, perms, &parent);
-  if (r) {
-    _ll_forget(*inode, 1);  
-    return r;
-  }
-
-  ceph_assert(parent != NULL);
-
-  // Num3: Finally, get the name (dentry) of the requested inode
-  r = _lookup_name(*inode, parent, perms);
-  if (r) {
-    // Unexpected error
-    _ll_forget(parent, 1);
-    _ll_forget(*inode, 1);
-    return r;
-  }
-
-  _ll_forget(parent, 1);
-  return 0;
```

Comment:
Please see d2ce029409689b1, and it seems trying to populating the cache with its parent dentry.

Reply:
Just one enhancement code...

Reply:
Thanks. Is that strictly required? Can it not populate the cache later, when the dentry is actually needed?

Reply:
Yeah, AFAIK, it should be okay. Checked it and I didn't see it will break any test or tool.

Reply:
Good, that was my interpretation too. For ganesha, we may not ever need to fetch the dentry info on certain workloads. For example, when a file is already open by the NFS client and ganesha is restarted, the filehandle is already known to the NFS client, so all it needs is the inode at that point. If it needs the dentry later, then it can deal with that then.

Both _lookup_parent and _lookup_name represent a call to the MDS, so this should reduce MDS/client traffic for ganesha overall.